server.go

```diff
@@ -898,6 +898,45 @@ func (s *Server) ReadMetricSocket(serverConn net.PacketConn, packetPool *sync.Po
 	}
 }
 
+// ReadStatsdDatagramSocket reads statsd metrics packets from connection off a unix datagram socket.
+func (s *Server) ReadStatsdDatagramSocket(serverConn *net.UnixConn, packetPool *sync.Pool) {
```

Comment:
I think you can re-use `ReadMetricSocket` instead of this new function, if you rejigger its interface a bit to correctly report "UDP" and "UNIX domain dgram socket" in error messages (:

Reply:
Tried to do that , but had to change a lot including changing the method signature to accept `interface {}` as one needs as `net.UnixConn` and other needs `net.PacketConn` and have to write `switch` type logic inside  and also requires changing `udpProcessor` type that's used inside `startSSFUDP` and `startStatsdUDP` 

So instead just extracted the common part to seperate method and DRY'ed up.
```
// ReadMetricSocket listens for available packets to handle.
func (s *Server) ReadMetricSocketS(network string, serverConn interface{}, packetPool *sync.Pool) {
	var packetConn net.PacketConn
	var unixConn net.UnixConn
	switch connType := serverConn.(type) {
	case net.PacketConn:
		packetConn = connType
	case net.UnixConn:
		unixConn = connType
	default:
		panic(fmt.Sprintf("Doesn't support  %v: only PacketConn and UnixConnection:// are supported", connType))

	}
.....
```

Reply:
Hm, thinking about this, it makes sense for all these handler functions to be distinct because the connection establishment / socket sharing logic is different also. I'm OK with this solution!